### PR TITLE
news_r3k.cpp: Add support for the NWS-3410

### PR DESCRIPTION
**Pull request by briceonk**
This will probably need some work before merging, but opening it up for review early since it is touching pre-existing code. This refactors the news_r3k driver so that the LCD panel is only present on the laptop models (for now, just the 3260 but hopefully more like the 3150 bigNEWS will show up some day :)) and adds the ROM metadata for the 3410. It boots to the monitor ROM just fine, but NetBSD hangs during SCSI init :(

One thing I need to check with the owner of the machine (and will update on this PR once I hear back) is if the IDROM dump is available. I tested this with the 3260 IDROM patched to look like a 3410.

### CODE REVIEW

**Review by cuavas — COMMENTED** [review]
I’m pretty tired at the moment, but I can’t see anything badly wrong with this at a glance.  @galibert and/or @pmackinlay do you want to take a look?

**Comment by galibert** [comment]
That looks very clean to me.

**Comment by pmackinlay** [comment]
Yes it's excellent and I'm happy to merge as soon as we have confirmed the name to credit.

**Comment by briceonk** [comment]
The dump credit should go to `gerdavax` (cc @gerdavax :)). Working on the ID-ROM stuff, which should be the last thing needed for this PR. Will update soon.

**Comment by rb6502** [comment]
@briceonk What's the status on the ID-ROM?  We're getting towards the freeze on Saturday and it'd be great to have this in.

**Comment by briceonk** [comment]
> @briceonk What's the status on the ID-ROM?  We're getting towards the freeze on Saturday and it'd be great to have this in.

Still waiting for the dump, unfortunately.

**Comment by cuavas** [comment]
@briceonk this PR is already conflicting with master.  Can we get the code changes in and add the IDPROM later when the dump becomes available?

**Comment by briceonk** [comment]
> @briceonk this PR is already conflicting with master.  Can we get the code changes in and add the IDPROM later when the dump becomes available?

I just so happened to get the IDPROM data this morning, so I will fix the merge conflicts and get the new metadata in this morning :)

**Comment by briceonk** [comment]
> @briceonk this PR is already conflicting with master. Can we get the code changes in and add the IDPROM later when the dump becomes available?

OK, this should be ready! I fixed the merge conflicts and added the correct metadata. I'll send the bin file in a bit. Thanks to @gerdavax for getting that data!

image

File: src/mame/sony/news_r3k.cpp [hunk 1]
```diff
@@ -523,8 +608,18 @@ ROM_START(nws3260)
...
+ROM_END
+
 } // anonymous namespace
 
 
-/*   YEAR  NAME     PARENT  COMPAT  MACHINE  INPUT    CLASS           INIT         COMPANY  FULLNAME    FLAGS */
-COMP(1991, nws3260, 0,      0,      nws3260, nws3260, news_r3k_state, init_common, "Sony",  "NWS-3260", MACHINE_NO_SOUND)
+/*   YEAR  NAME     PARENT  COMPAT  MACHINE  INPUT    CLASS          INIT         COMPANY  FULLNAME    FLAGS */
+COMP(1991, nws3260, 0,      0,      nws3260, nws3260, nws3260_state, init_common, "Sony",  "NWS-3260", MACHINE_NO_SOUND)
+COMP(1991, nws3410, 0,      0,      nws3410, nws3410, nws3410_state, init_common, "Sony",  "NWS-3410", MACHINE_NO_SOUND)
```
Comment (cuavas) [inline src/mame/sony/news_r3k.cpp:625]: I’ve already merged this before it rots again, but should this system have an unimplemented graphics flag due to lack of framebuffer emulation?  I can add the flag myself.

Comment (briceonk) [reply src/mame/sony/news_r3k.cpp:625]: Oh, yeah, it should. Sorry, I must have forgotten to add that myself :(

**Pull request merged**